File: problems/leetcode/997-find_the_town_judge.c
#include <stdlib.h>

int findJudge(int n, int** trust, int trustSize, int* trustColSize) {
    int* trustCount = (int*)calloc(n + 1, sizeof(int)); 
    int* trustedByCount = (int*)calloc(n + 1, sizeof(int)); 
    
    for (int i = 0; i < trustSize; i++) {
        int a = trust[i][0];
        int b = trust[i][1];
        trustCount[b]++;
        trustedByCount[a]++;
    }
    
    for (int i = 1; i <= n; i++) {
        if (trustCount[i] == n - 1 && trustedByCount[i] == 0) {
            free(trustCount);
            free(trustedByCount);
            return i; 
        }
    }
    
    free(trustCount);
    free(trustedByCount);
    return -1; 
}

File: solutions/leetcode/997-find_the_town_judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        trust_count = [0] * (n + 1) 
        trusted_by_count = [0] * (n + 1) 
        
        for a, b in trust:
            trust_count[b] += 1
            trusted_by_count[a] += 1
        
        for i in range(1, n + 1):
            if trust_count[i] == n - 1 and trusted_by_count[i] == 0:
                return i  
        
        return -1  